client/tso_dispatcher.go

```diff
@@ -110,23 +110,23 @@ func (req *tsoRequest) Wait() (physical int64, logical int64, err error) {
 	}
 }
 
-func (c *client) updateTSODispatcher() {
+func (c *tsoClient) updateTSODispatcher() {
 	// Set up the new TSO dispatcher and batch controller.
-	c.tsobc.GetTSOAllocators().Range(func(dcLocationKey, _ interface{}) bool {
+	c.GetTSOAllocators().Range(func(dcLocationKey, _ interface{}) bool {
 		dcLocation := dcLocationKey.(string)
 		if !c.checkTSODispatcher(dcLocation) {
 			c.createTSODispatcher(dcLocation)
 		}
 		return true
 	})
 	// Clean up the unused TSO dispatcher
-	c.tsoDispatcher.Range(func(dcLocationKey, _ interface{}) bool {
+	c.tsoDispatcher.Range(func(dcLocationKey, dispatcher interface{}) bool {
 		dcLocation := dcLocationKey.(string)
 		// Skip the Global TSO Allocator
 		if dcLocation == globalDCLocation {
 			return true
 		}
-		if dispatcher, exist := c.tsobc.GetTSOAllocators().Load(dcLocation); !exist {
+		if _, exist := c.GetTSOAllocators().Load(dcLocation); !exist {
 			log.Info("[pd/tso] delete unused tso dispatcher", zap.String("dc-location", dcLocation))
 			dispatcher.(*tsoDispatcher).dispatcherCancel()
```

Comment:
Without the change in updateTSODispatcher, L131 returned dispatcher isn't the type of (*tsoDispatcher). This issue is legacy issue and is now uncovered after refactor enables more test paths. It means the block between L129 and L132 was never touched before.